README.md

```diff
@@ -10,4 +10,21 @@ The redesign of OSHI intends to achieve:
 - A caching policy to improve performance
 - Generic code for fetching that can be reused
 - Usage similar to OSHI 4
-- Runtime dependence on [JNA](https://github.com/java-native-access/jna) only
\ No newline at end of file
+- Runtime dependence on [JNA](https://github.com/java-native-access/jna) only
+
+## [Terminology](https://github.com/oshi/oshi5/issues/7)
+`// TODO`
+
+## Design Decisions
+### [Code Generation](https://github.com/oshi/oshi5/issues/4)
+OSHI's API layer will be automatically generated from a central YAML definitions file. YAML is great for representing tree structures (like OSHI's API) and has a much simpler syntax than JSON or XML. This approach will ensure the resulting API is absolutely uniform, synchronized at all times, and will greatly simplify API changes.
+
+The generator runs on every build (via Maven integration) and updates source files in the API layer's packages according to the central definitions file. Generated files will be committed to the repository.
+
+The definitions file can also be converted into supplemental documentation that contains every feature's description and compatibility requirements in a single place. This addresses the concern brought up [in this comment](https://github.com/oshi/oshi5/issues/2#issuecomment-451220174).
+
```

Comment:
Looks great as a concept but lacking in details.  Need a couple sentences/bullets outlining what information will be included in the YAML.   In addition to whatever we'll use to represent the name/path/package/json tree/whatever we call it, need to specify the documentation here.

I say this having dealt today with an auto-generated bare-bones API that was utterly useless.

Reply:
I just added an initial spec for the definitions YAML.
> In addition to whatever we'll use to represent the name/path/package/json tree/whatever we call it, need to specify the documentation here.

I haven't figured out the best way to do this yet.
- Maybe we could have a `package` field applicable to the top-level types in the YAML that describes how users will navigate the API tree to obtain those objects.
- Or just make the top-level types the full paths similar to a Java class. As an example, `Disk` would be `Computer.Hardware.Disk`.

The second option is starting to sound cleaner to me.